feature-libs/checkout/core/facade/checkout-payment.service.ts

```diff
@@ -121,7 +122,7 @@ export class CheckoutPaymentService implements CheckoutPaymentFacade {
         this.checkoutStore.dispatch(
           new CheckoutActions.SetPaymentDetails({
             userId,
-            cartId: cart.code,
+            cartId: cart.code ?? '',
```

Comment:
We could also change the type in the action to have cartId typed as `string | undefined`.  Although I don't know if we want to do that.